src/Compilers/CSharp/Portable/Symbols/Source/ExtensionGroupingInfo.cs

```diff
@@ -184,6 +184,33 @@ internal IEnumerable<ImmutableArray<SourceNamedTypeSymbol>> EnumerateMergedExten
             }
         }
 
+        /// <summary>
+        /// Reports a diagnostic when two extension blocks grouped into a single marker type have different C#-level signatures.
+        /// Note: It's fine for two extension blocks with different IL-level signatures to be grouped into the same grouping type, so we don't check for collisions there.
```

Comment:
>so we don't check for collisions there [](http://example.com/codeflow?start=131&length=38)

It is not clear whether a statement is made that collisions in grouping type are impossible to occur. If it is, I think it is incorrect and collisions checks for grouping types are also necessary.